File: src/tf_idf.c
#include <stddef.h>
#include <stdio.h>
#include <math.h>

double tf(size_t nmbr_of_t_appearence, size_t nmbr_of_wrds){
    return (double)nmbr_of_t_appearence/(double)nmbr_of_wrds;
}

double idf(size_t nmbr_of_pgs, size_t nmbr_of_pgs_where_t){
    return log((double)nmbr_of_pgs/(double)nmbr_of_pgs_where_t);
}

double tf_idf(size_t a, size_t b, size_t c, size_t d){
    double calc_tf = tf(a, b);
    double calc_idf = idf(c, d);
    return calc_tf * calc_idf;
}

File: src/tf_idf.rs
pub fn tf_idf(n_t_appr_in_d: f64, n_t_words_in_d: f64, n_d: f64, n_d_where_t: f64) -> f64 {
    let tf: f64 = n_t_appr_in_d / n_t_words_in_d;
    let idf: f64 = (n_d / n_d_where_t).log(10.0);
    tf * idf
}